cmd/config-reloader/main.go

```diff
@@ -87,9 +87,10 @@ func main() {
 				os.Exit(0)
 			case <-ticker.C:
 				resp, err := http.DefaultClient.Do(req)
+				// Log but tolerate the error here as a "connection refused"
+				// error from Prometheus is a retryable event during startup.
 				if err != nil {
 					level.Error(logger).Log("msg", "polling ready-url", "err", err)
-					os.Exit(1)
```

Comment:
@pintohutch do we maybe want to log non-OK statuses in the same spirit as we log here? In that case, would we maybe want to `break` here?

As an aside, I'm not sure why we're using a goroutine here -- we could have just `break`ed the infinite `for` loop within the `select`.